fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvPartitionCompensator.java

```diff
@@ -0,0 +1,696 @@
+                                                              OptExpression queryExpression) {
+        List<LogicalScanOperator> scanOperators = MvUtils.getScanOperator(queryExpression);
+        if (scanOperators.isEmpty()) {
+            return ConstantOperator.createBoolean(true);
+        }
+
+        List<ScalarOperator> partitionPredicates = Lists.newArrayList();
+        boolean isCompensatePartition = mvContext.getOrInitCompensatePartitionPredicate(queryExpression);
+        // Compensate partition predicates and add them into query predicate.
+        Map<Pair<LogicalScanOperator, Boolean>, List<ScalarOperator>> scanOperatorScalarOperatorMap =
+                mvContext.getScanOpToPartitionCompensatePredicates();
+        for (LogicalScanOperator scanOperator : scanOperators) {
+            if (!SUPPORTED_PARTITION_COMPENSATE_SCAN_TYPES.contains(scanOperator.getOpType())) {
+                continue;
+            }
+            List<ScalarOperator> partitionPredicate = scanOperatorScalarOperatorMap
+                    .computeIfAbsent(Pair.create(scanOperator, isCompensatePartition), x -> {
+                        return isCompensatePartition ? getCompensatePartitionPredicates(columnRefFactory, scanOperator) :
+                                getScanOpPrunedPartitionPredicates(scanOperator);
+                    });
+            if (partitionPredicate == null) {
+                return null;
+            }
+            partitionPredicates.addAll(partitionPredicate);
+        }
+        return partitionPredicates.isEmpty() ? ConstantOperator.createBoolean(true) :
+                Utils.compoundAnd(partitionPredicates);
+    }
+
+    private static List<ScalarOperator> getCompensatePartitionPredicates(ColumnRefFactory columnRefFactory,
+                                                                         LogicalScanOperator scanOperator) {
+        List<ScalarOperator> partitionPredicate = null;
+        if (scanOperator instanceof LogicalOlapScanOperator) {
+            partitionPredicate = compensatePartitionPredicateForOlapScan((LogicalOlapScanOperator) scanOperator,
+                    columnRefFactory);
+        } else if (SUPPORTED_PARTITION_COMPENSATE_EXTERNAL_SCAN_TYPES.contains(scanOperator.getOpType())) {
+            partitionPredicate = compensatePartitionPredicateForExternalTables(scanOperator);
+        } else {
+            return null;
+        }
+        return partitionPredicate;
+    }
+
+
+    private static boolean isSupportedPartitionPredicateForExternalTables(List<PartitionKey> partitionKeys) {
+        for (PartitionKey partitionKey : partitionKeys) {
+            // only support one partition column now.
+            if (partitionKey.getKeys().size() != 1) {
+                return false;
+            }
+            LiteralExpr e = partitionKey.getKeys().get(0);
+            // Only support date/int type
+            if (!(e instanceof DateLiteral || e instanceof IntLiteral)) {
```

Comment:
what about other types, like string, datetime

Reply:
remove this check.